gluoncv/auto/estimators/image_classification/image_classification.py

```diff
@@ -224,30 +230,45 @@ def _train_loop(self, train_data, val_data, time_limit=math.inf):
-
-            top1_val, top5_val = self._evaluate(val_data)
-            early_stopper.update(top1_val)
-
+            
             self._logger.info('[Epoch %d] training: %s=%f', epoch, train_metric_name, train_metric_score)
             self._logger.info('[Epoch %d] speed: %d samples/sec\ttime cost: %f', epoch, throughput, time.time()-tic)
-            self._logger.info('[Epoch %d] validation: top1=%f top5=%f', epoch, top1_val, top5_val)
-
-            if top1_val > self._best_acc:
-                cp_name = os.path.join(self._logdir, _BEST_CHECKPOINT_FILE)
-                self._logger.info('[Epoch %d] Current best top-1: %f vs previous %f, saved to %s',
-                                  self.epoch, top1_val, self._best_acc, cp_name)
-                self.save(cp_name)
-                self._best_acc = top1_val
-            if self._reporter:
-                self._reporter(epoch=epoch, acc_reward=top1_val)
+            
+            val_score = self._evaluate(val_data, metric_name=train_metric_name)
+            if train_metric_name == 'rmse':
+                early_stopper.update(val_score)
+                self._logger.info('[Epoch %d] validation: rmse=%f', epoch, val_score)
+                if val_score > self._best_acc:
+                    cp_name = os.path.join(self._logdir, _BEST_CHECKPOINT_FILE)
+                    self._logger.info('[Epoch %d] Current best rmse: %f vs previous %f, saved to %s',
+                                      self.epoch, val_score, self._best_acc, cp_name)
+                    self.save(cp_name)
+                    self._best_acc = val_score
+                if self._reporter:
+                    self._reporter(epoch=epoch, acc_reward=val_score)
+            else:
+                top1_val, top5_val = val_score
+                early_stopper.update(top1_val)
+                self._logger.info('[Epoch %d] validation: top1=%f top5=%f', epoch, top1_val, top5_val)
+
+                if top1_val > self._best_acc:
+                    cp_name = os.path.join(self._logdir, _BEST_CHECKPOINT_FILE)
+                    self._logger.info('[Epoch %d] Current best top-1: %f vs previous %f, saved to %s',
+                                      self.epoch, top1_val, self._best_acc, cp_name)
+                    self.save(cp_name)
+                    self._best_acc = top1_val
+                if self._reporter:
+                    self._reporter(epoch=epoch, acc_reward=top1_val)
             self._time_elapsed += time.time() - post_tic
         return {'train_acc': train_metric_score, 'valid_acc': self._best_acc,
                 'time': self._time_elapsed, 'checkpoint': cp_name}
 
     def _init_network(self):
-        if not self.num_class:
-            raise ValueError('Unable to create network when `num_class` is unknown. \
+        if not self.num_class and self._problem_type != REGRESSION:
+            raise ValueError('This is not a regression problem. Unable to create network when `num_class` is unknown. \
```

Comment:
Change to `This is a classification problem and we are not able to create the network when `num_class` is unknown`.